labs/src/main/scala/org/scalalabs/basic/lab03/RecursionPatternMatchingExercise.scala

```diff
@@ -29,39 +29,59 @@ object RecursionPatternMatchingExercise {
    * checkValuesIncrease(Seq(1,2,2)) == false
    */
   def checkValuesIncrease(seq: Seq[Int]): Boolean = {
-    error("fix me")
+    if (seq.length == 0) return false
+    if (seq.length == 1) return true
+    else
+      (seq(0) < seq(1)) match {
+      case true => return checkValuesIncrease(seq.tail)
+      case false => return false
+    }
   }
   
   /**
    * Group Consecutive values
    * List(1,1,2,3,1,1) -> List(1,1), List(2), List(3), List(1,1)
    */
-  def groupConsecutive[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+  def groupConsecutive[T](in: List[T]): List[List[T]] = in.length match{
+    case 0 => List()
+    case 1 => List(in)
+    case _ => subGroup(in) :: groupConsecutive(in.drop(subGroup(in).length))
+  }
+  
+  def subGroup[T](in: List[T]): List[T] = in.length match {
+    case 0 => List()
+    case 1 => in
+    case _ => in takeWhile {in.head ==}
   }
 
   /**
    * Group Equal values
    * List(1,1,2,3,1,1) -> List(1,1,1,1), List(2), List(3)
    */
-  def groupEquals[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+  def groupEquals[T](in: List[T]): List[List[T]] = in.length match{
+    case 0 => List()
+    case 1 => List(in)
+    case _ => List(in.filter((item: T) => item == in.head)) ::: groupEquals(in.filter((item: T) => item != in.head))
```

Comment:
I don't think it was required here, but you can match on the list itself, not just the length:

```
def length[T](list: Seq[T]) = list match {
    case Nil => 0
    case _:rest => 1 + length(rest)
}
```